[cpu] Making aminmax and clamp work with half dtype

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100010

Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @soumith @desertfire

HDCharles committed:
[cpu] Making aminmax and clamp work with half dtype

Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. Also desirable for gpu int8 quantization so we can use aminmax
to traverse tensor once to get min/max.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. Also desirable for gpu int8 quantization so we can use aminmax
to traverse tensor once to get min/max.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. Also desirable for gpu int8 quantization so we can use aminmax
to traverse tensor once to get min/max.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

mingfeima reviewed (approved):
LGTM, just need to clear the CI failures.

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. Also desirable for gpu int8 quantization so we can use aminmax
to traverse tensor once to get min/max.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

HDCharles committed:
Update on "[cpu] Making aminmax and clamp work with half dtype"


Summary: This is helpful for debugging/comparing gpu numerics where fp16
is extremely common. note: some other ops were affected by this change,
enabling them to run on torch.half as well. Where necessary the op
info/tests were changed to avoid looking for errors where they no longer
occur.

Test Plan: python test/test_reductions.py
python test/test_shape_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]
